### PR TITLE
 Add events page to ceph storage dashboard

### PR DESCRIPTION
**Pull request by cloudbehl**
Screenshot from 2019-07-08 15-29-50

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import {
+  EventModel,
+  PersistentVolumeClaimModel,
+  PersistentVolumeModel,
+} from '@console/internal/models';
+import { FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { EventKind } from '@console/internal/module/k8s';
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx]: empty line between `@console/internal/module/k8s` and ../../../constants/index

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { cephStorageNamespace } from '../../../constants/index';
+
+const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
+
+const ocsEventNamespaceFilter = (event: EventKind): boolean =>
+  _.get(event, 'metadata.namespace') === cephStorageNamespace ||
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx]: you could reuse `getNamespace` from `@console/shared`

Comment (rawagner) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx]: why not to put the body of function directly into `ocsEventFilter` ? Or split the function to `ocsEventNamespaceFilter` and `ocsEventKindFilter`

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Events</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <EventsBody
+          events={resources.events as FirehoseResult<EventKind[]>}
+          filter={(event) => ocsEventFilter(event)}
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/events-card.tsx]:
```
filter={ocsEventFilter}
```

File: frontend/packages/ceph-storage-plugin/src/constants/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const cephStorageNamespace = 'openshift-storage';
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/constants/index.ts]: `CEPH_STORAGE_NAMESPACE`

**Comment by cloudbehl** [comment]
@rawagner updated the PR

**Comment by rawagner** [comment]
@cloudbehl can you rebase ?

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 5]
```diff
@@ -47,6 +47,17 @@ const plugin: Plugin<ConsumedExtensions> = [
...
     },
   },
+  {
+    type: 'Dashboards/Card',
+    properties: {
+      tab: 'persistent-storage',
+      position: GridPosition.RIGHT,
+      loader: () =>
+        import(
+          './components/dashboard-page/storage-dashboard/events-card' /* webpackChunkName: "ceph-data-resiliency-card" */
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts]:
```suggestion
          './components/dashboard-page/storage-dashboard/events-card' /* webpackChunkName: "ceph-events-card" */
```

**Pull request merged**